From 09e7e3cd5956efa9bebebeb3e8fcbcb3e5a4290a Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Thu, 16 Nov 2017 15:33:43 -0800
Subject: [PATCH] joyent/node-watershed#2 Use crypto.randomBytes() for key
 generation Reviewed by: Cody Peter Mello <cody.mello@joyent.com> Approved by:
 Joshua M. Clulow <jmc@joyent.com>

---
 CHANGES.md       |  1 +
 lib/watershed.js | 11 ++---------
 2 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/CHANGES.md b/CHANGES.md
index 9651e6f..a449449 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -6,6 +6,7 @@
 
 * Support for Websocket sub-protocol negotiation
 * Documented `detached`/raw socket argument to `accept()`
+* Change to using `crypto.randomBytes()` for entropy
 
 ## 0.3.4
 
diff --git a/lib/watershed.js b/lib/watershed.js
index d55dedb..5514c2b 100644
--- a/lib/watershed.js
+++ b/lib/watershed.js
@@ -117,11 +117,7 @@ Watershed()
 Watershed.prototype.generateKey = function
 generateKey()
 {
-	var nonce = new Buffer(NONCE_LENGTH);
-	for (var i = 0; i < nonce.length; i++) {
-		nonce[i] = Math.floor(Math.random() * 256);
-	}
-	return (nonce.toString('base64'));
+	return (crypto.randomBytes(NONCE_LENGTH).toString('base64'));
 };
 
 /*
@@ -597,10 +593,7 @@ _ws_writeFrameCommon(opcode, data)
 	 * According to the RFC, the client MUST mask their outgoing frames.
 	 */
 	if (this._options.localShouldMask) {
-		maskbuf = new Buffer(4);
-		for (var i = 0; i < maskbuf.length; i++) {
-			maskbuf[i] = Math.floor(Math.random * 256);
-		}
+		maskbuf = crypto.randomBytes(4);
 		for (var j = 0; j < data.length; j++) {
 			data[j] = data[j] ^ maskbuf[j % maskbuf.length];
 		}
-- 
2.21.0

